StardewArchipelago/Locations/CodeInjections/Modded/SVE/SVECutsceneInjections.cs

```diff
@@ -96,27 +96,25 @@ public static bool CheckForAction_LanceChest_Prefix(Chest __instance, Farmer who
             }
         }
 
-        public static void ResetLocalState_PlayRailroadBoulderCutsceneIfConditionsAreMet_Postfix(GameLocation __instance)
+        public static bool EndBehaviors_AddRailroadBoulderIfIridiumBomb_Prefix(string[] split, Event __instance)
         {
             try
             {
-                var bombBeforeQuest = Game1.player.eventsSeen.Contains(RAILROAD_BOULDER_ID) && Game1.player.eventsSeen.Contains(IRIDIUM_BOMB_ID);
-                if (!Game1.player.hasSkullKey || !bombBeforeQuest)
+                if (__instance.id != RAILROAD_BOULDER_ID || !Game1.player.mailReceived.Contains("RailroadBoulderRemoved"))
                 {
-                    return;
+                    return true;
```

Comment:
please add the `// run original logic` and `// don't run original logic` comments next to returns in a patch, to signify that it is not the return value, but the desired vanilla behavior being affected or not